src/DurableTask.Netherite/Scaling/AzureTableLoadPublisher.cs

```diff
@@ -13,104 +16,84 @@ namespace DurableTask.Netherite.Scaling
+            catch(Exception) // table may not exist
             {
-                var batch = await this.table.ExecuteQuerySegmentedAsync<PartitionInfoEntity>(query, continuationToken, null, null, cancellationToken).ConfigureAwait(false);
 
-                if (batch.Count() > 0)
-                {
-                    // delete all entities in this batch. Max partition number is 32 so it always fits.
-                    TableBatchOperation tableBatch = new TableBatchOperation();
-
-                    foreach (var e in batch)
-                    {
-                        tableBatch.Add(TableOperation.Delete(e));
-                    }
-
-                    await this.table.ExecuteBatchAsync(tableBatch).ConfigureAwait(false);
-                }
             }
-            while (continuationToken != null);
         }
 
-        public async Task CreateIfNotExistsAsync(CancellationToken cancellationToken)
+        public Task CreateIfNotExistsAsync(CancellationToken cancellationToken)
         {
-            if (await this.table.ExistsAsync().ConfigureAwait(false))
-            {
-                return;
-            }
-
-            await this.table.CreateAsync().ConfigureAwait(false);
+            return this.table.CreateIfNotExistsAsync();
         }
 
         public Task PublishAsync(Dictionary<uint, PartitionLoadInfo> info, CancellationToken cancellationToken)
         {
-            TableBatchOperation tableBatch = new TableBatchOperation();
+            var tableBatch = new List<TableTransactionAction>();          
             foreach(var kvp in info)
             {
-                tableBatch.Add(TableOperation.InsertOrReplace(new PartitionInfoEntity(this.taskHubName, kvp.Key, kvp.Value)));
+                tableBatch.Add(new TableTransactionAction(TableTransactionActionType.UpsertReplace, new PartitionInfoEntity(this.taskHubName, kvp.Key, kvp.Value)));
             }
-            return this.table.ExecuteBatchAsync(tableBatch, null, null, cancellationToken);
+            return this.table.SubmitTransactionAsync(tableBatch, cancellationToken);
         }
 
         public async Task<Dictionary<uint, PartitionLoadInfo>> QueryAsync(CancellationToken cancellationToken)
         {
-            var query = new TableQuery<PartitionInfoEntity>().Where(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, this.taskHubName));
-            TableContinuationToken continuationToken = null;
             Dictionary<uint, PartitionLoadInfo> result = new Dictionary<uint, PartitionLoadInfo>();
-            do
+            await foreach (var e in this.table.QueryAsync<PartitionInfoEntity>($"PartitionKey eq '{this.taskHubName}'"))
             {
-                var batch = await this.table.ExecuteQuerySegmentedAsync<PartitionInfoEntity>(query, continuationToken, null, null, cancellationToken).ConfigureAwait(false);
-                foreach (var e in batch)
+                int.TryParse(e.CachePct, out int cachePct);
```

Comment:
is it fine if this and the double fail to parse?

Reply:
There is no good way to log errors from this code, so the calling code has to catch and report exceptions. Same is true for other exceptions.

Reply:
yeah, the exception policy is not very consistent actually. Also, this class should really have unit tests, I think there are bugs here.